pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -651,3 +706,118 @@ func getMinTime(t1, t2 time.Duration) time.Duration {
 	}
 	return t2
 }
+
+func (exp *FlowExporter) l7Listener() {
+	if err := os.Remove(exp.l7ListenerSocketPath); err != nil && !os.IsNotExist(err) {
+		klog.ErrorS(err, "Error removing Suricata socket")
+	}
+	if err := os.MkdirAll(filepath.Dir(exp.l7ListenerSocketPath), 0755); err != nil {
+		klog.Fatalf("Failed to create directory %s: %v", filepath.Dir(exp.l7ListenerSocketPath), err)
+	}
+	listener, err := net.Listen("unix", exp.l7ListenerSocketPath)
+	if err != nil {
+		klog.ErrorS(err, "Error listening on Suricata socket")
+		return
+	}
+	defer listener.Close()
+
+	klog.InfoS("L7 Listener Server started. Listening for connections...")
+
+	for {
+		conn, err := listener.Accept()
+		if err != nil {
+			klog.ErrorS(err, "Error accepting Suricata connection")
+			continue
+		}
+		go func() {
+			exp.handleClientConnection(conn)
+		}()
+	}
+}
+
+func (exp *FlowExporter) handleClientConnection(conn net.Conn) {
+	defer conn.Close()
+	reader := bufio.NewReader(conn)
+	for {
+		buffer, err := reader.ReadBytes('\n')
+		if err != nil && err != io.EOF {
+			klog.ErrorS(err, "Error reading data", "buffer", buffer)
+			return
+		}
+		exp.processLog(buffer)
+	}
+}
+
+func (exp *FlowExporter) processLog(data []byte) {
+	exp.l7mut.Lock()
+	defer exp.l7mut.Unlock()
+	// Check if the event type is "http"
+	if eventType := extractEventType(data); eventType != "http" {
+		return
+	}
+	var event JsonToEvent
+	err := json.Unmarshal(data, &event)
+	if err != nil {
+		klog.ErrorS(err, "Error parsing JSON:", "data", data)
+		return
+	}
+	exp.WriteConnection(event)
+}
+
+func extractEventType(data []byte) string {
+	// Define a temporary struct for extracting the event type
+	type Temp struct {
+		EventType string `json:"event_type"`
+	}
+
+	// Parse the JSON string and extract the event type
+	var temp Temp
+	err := json.Unmarshal(data, &temp)
+	if err != nil && !strings.Contains(err.Error(), "unexpected end of JSON input") {
+		klog.ErrorS(err, "Error parsing JSON for eventtype:", "data", data)
+		return ""
+	}
+
+	return temp.EventType
+}
+
+func (exp *FlowExporter) WriteConnection(event JsonToEvent) {
```

Comment:
The function name is too general, and also why it has to be public? something like `addOrUpdateL7Event` would be better?

Reply:
I agree!, Thanks for pointing out!